### ISSUE TITLE
[manuscript dev] Improve handling of missing manuscripts

### ISSUE THREAD

**Issue by PeterBjuhr** [wbsoft/frescobaldi#756]
The first step for this has already been taken so I would like to complete this.

Currently a file is silently removed by default if it can't be found in the given path. 

When reloading files through a session the user is additionally notified that the files are missing.

I think at least the user should be notified in all cases but possibly there are more things to be done.

See also the discussion in e72c415925e6d023720b3d48871a69661d320c7c.

**Comment by PeterBjuhr** [comment]
I'm considering to let the user decide whether to remove the missing file or try to restore it manually; I think there are use cases where you would rather leave the missing file in the drop-down list/`ViewdocChooser` and restore it than to reopen it through the file dialog. For example if you by mistake moved a folder with a lot of manuscripts it would probably be easier to restore the folder than to reopen all the files.

Is this a good idea?

**Comment by uliska** [comment]
Well, this was more or less my first suggestion so obviously I find this a good idea ;-)

**Comment by PeterBjuhr** [comment]
Great. By the way, when did you suggest that? I may have missed that...

**Comment by uliska** [comment]
I think you *dis*missed that then (as overly complex).

There must be a closed issue about the topic where I suggested that, presumably in the initial comment (can't search for that right now).

**Comment by PeterBjuhr** [comment]
This is perhaps a minor detail but I want to avoid misunderstandings. I think you are referring to #738 (by the way I realize now that I could have reopened that instead but I think it's better with a new issue...)

You were suggesting:

>an interface to reopen [the missing file]

To be clear I am only suggesting to give the user the choice to avoid the document being removed (which is done now by default). All the restoring has to be done externally.

**Comment by uliska** [comment]
OK. You mean actually using the same dialog as we have now, listing the
missing files, but present the appropriate selection buttons for it?

I'd be fine with that.

**Comment by PeterBjuhr** [comment]
Yes, only a dialog. I'm also considering adding the option for the user to save the choice for future reference ('always do this...').

**Comment by uliska** [comment]
Seems like a good idea, but: we will need the dialog each time again to
inform the user about a missing file. So there might not be that much
use in such a preference.

**Comment by PeterBjuhr** [comment]
I was thinking about two checkboxes, something like:

- [ ] Always inform me about missing files
- [ ] Always do this action

That way the user could also avoid the dialog. What do you think?

**Comment by uliska** [comment]
I would have to see it in action to make an opinion. My gut feeling says that the case of missing files is a rather unusual case and will happen only rarely. That would mean that it would be better to always inform the user about the instance.

**Comment by PeterBjuhr** [comment]
Yes, I think you're right, missing files could be expected to be rather rare (that was also my point before about checking every file if it exists). I'll start with a simple dialog and we'll take it from there.

**Comment by uliska** [comment]
There's important discussion in https://github.com/wbsoft/frescobaldi/commit/c1ac033bf59eabc5b2aeab70470a4899807c905f
